proto/toolchain.bzl

```diff
@@ -0,0 +1,58 @@
+
+def _emit_proto_compile(ctx, 
+    proto_toolchain, go_proto_toolchain, 
+    name, lib, importpath):
+  go_srcs = []
+  outpath = None
+  for proto in lib.proto.direct_sources:
+    out = ctx.new_file(importpath + "/"+ proto.basename[:-len(".proto")] + ".pb.go")
+    go_srcs += [out]
+    if outpath == None:
+        outpath = out.dirname[:-len(importpath)]
+  args= [
+      "--go_out=" + ",".join(go_proto_toolchain.options) + ":"+outpath,
+      "--plugin=protoc-gen-go=" + go_proto_toolchain.plugin.path,
+      "--descriptor_set_in", ":".join(
+          [s.path for s in lib.proto.transitive_descriptor_sets])
+  ]
+  args += [proto.short_path for proto in lib.proto.direct_sources]
+  ctx.action(
```

Comment:
Why `command` and not `executable` + `arguments`?

Reply:
Because it was originally but I changed it so I could append `"&& tree"`, it's the only way to debug rules that don't produce the expected outputs, and I forgot to change it back again when I was done :)